From ab9e934b63ef3b986525c57bc27a3c805d79c870 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 31 Mar 2022 21:54:57 +0300
Subject: [PATCH] [SW-77251] gaudi3: add global register init

This patch perform global register initialization which is used
by firmware update module.

Change-Id: If816471068c7727511ef309d1bc3db88f2224e2b
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index f73cf91..51cca2e 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -13,6 +13,7 @@
 #include <dev_common.h>
 #include <dev_efuse.h>
 #include <device.h>
+#include <gaudi3_reg_map.h>
 
 static struct cmd_module_ops cmd_ops[] = {
 	{	/* PCI module */
@@ -42,8 +43,21 @@
 	/* TODO: implement erom recovery */
 }
 
+static void device_init_global_conf(void)
+{
+	global_conf.upd_cmd_map = mmUPD_CMD;
+	global_conf.upd_sts_map = mmUPD_STS;
+}
+
+static void device_init(void)
+{
+	device_init_global_conf();
+}
+
 static void rom_recovery(void)
 {
+	device_init();
+
 	module_handle_cmds(cmd_ops, ARRAY_SIZE(cmd_ops));
 }
 
